ComponentKit/Utilities/CKComponentActionInternal.h

```diff
@@ -21,50 +21,74 @@
  We support several different types of action variants. You don't need to use this value anywhere, it's set for you
  by whatever initializer you end up using.
  */
-typedef NS_ENUM(NSUInteger, _CKTypedComponentActionVariant) {
-  _CKTypedComponentActionVariantRawSelector = 0,
-  _CKTypedComponentActionVariantTargetSelector,
-  _CKTypedComponentActionVariantComponentScope
+typedef NS_ENUM(NSUInteger, CKTypedComponentActionVariant) {
+  CKTypedComponentActionVariantRawSelector = 0,
+  CKTypedComponentActionVariantTargetSelector,
+  CKTypedComponentActionVariantComponentScope
+};
+
+typedef NS_ENUM(NSUInteger, CKComponentActionSendBehavior) {
+  /** Starts searching at the sender's next responder. Usually this is what you want to prevent infinite loops. */
+  CKComponentActionSendBehaviorStartAtSenderNextResponder,
+  /** If the sender itself responds to the action, invoke the action on the sender. */
+  CKComponentActionSendBehaviorStartAtSender,
+};
+
+struct CKTypedComponentActionValue {
+  CKTypedComponentActionValue() : _variant(CKTypedComponentActionVariantRawSelector), _target(nil), _scopeHandle(nil), _selector(NULL) {}
+  CKTypedComponentActionValue(const CKTypedComponentActionValue &value) : _variant(value._variant), _target(value._target), _scopeHandle(value._scopeHandle), _selector(value._selector) {};
```

Comment:
more inclined to outline this one than the default ctor. In addition to the 4 stores, it's got `objc_storeWeak` calls.